test-poc/framework/src/main/java/org/keycloak/test/framework/realm/RealmSupplier.java

```diff
@@ -41,6 +41,9 @@ public RealmResource getValue(InstanceContext<RealmResource, TestRealm> instance
 
         adminClient.realms().create(realmRepresentation);
 
+        // TODO Token needs to be invalidated after creating realm to have roles for new realm in the token. Maybe lightweight access tokens could help.
```

Comment:
Do we have an issue logged for this?

Reply:
https://github.com/keycloak/keycloak/issues/31513

Reply:
That's for supporting it in Keycloak, will add another to update the test poc